scripts/runfabtests.sh

```diff
@@ -2,23 +2,46 @@
 
 trap cleanup_and_exit SIGINT
 
+#
+# Default behavior with no args will use sockets provider with loopback
+#
 declare BIN_PATH
-declare PROV
+declare PROV="sockets"
 declare TEST_TYPE="quick"
-declare SERVER
-declare CLIENT
+declare SERVER="127.0.0.1"
+declare SSH_SERVER
+declare CLIENT="127.0.0.1"
+declare SSH_CLIENT
 declare EXCLUDE
 declare GOOD_ADDR="192.168.10.1"
 declare -i VERBOSE=0
+declare XARGS_CMD
+
+declare OSNAME=`uname -s | tr '[A-Z]' '[a-z]'`
+
+if [ -z "$TMPDIR" ]; then
+    TMPDIR=/tmp/
+fi
 
 # base ssh,  "short" and "long" timeout variants:
 declare -r bssh="ssh -n -o StrictHostKeyChecking=no -o ConnectTimeout=2 -o BatchMode=yes"
-declare -r sssh="timeout 60s ${bssh}"
-declare -r lssh="timeout 90s ${bssh}"
+if [ "$OSNAME" == "linux" ]; then
+    declare -r sssh="timeout 60s ${bssh}"
+    declare -r lssh="timeout 90s ${bssh}"
+    XARGS_CMD="xargs -r"
+elif [ "$OSNAME" == "darwin" ]; then
+    # No simple equivalent of timeout for OSX
+    declare -r sssh="${bssh}"
+    declare -r lssh="${bssh}"
+    # xargs -r behavior is default
+    XARGS_CMD="xargs"
+# else
+    # Could issue warning if we want
+fi
 declare ssh=${sssh}
 
-declare -r c_outp=$(mktemp)
-declare -r s_outp=$(mktemp)
+declare -r c_outp=$(mktemp $TMPDIR/fabtests.c_outp.XXXXXX)
+declare -r s_outp=$(mktemp $TMPDIR/fabtests.s_outp.XXXXXX)
```

Comment:
On reading the man pages, both versions use TMPDIR if it is defined, otherwise /tmp.  FWIW, TMPDIR is defined on my Mac.